rpcs3/Emu/Cell/lv2/sys_fs.h

```diff
@@ -216,12 +216,15 @@ struct lv2_dir final : lv2_fs_object
 	// Read next
 	const fs::dir_entry* dir_read()
 	{
-		if (const u64 cur = pos++; cur < entries.size())
+		return pos.fetch_op([this](u64& value) -> const fs::dir_entry*
```

Comment:
It was actually optimization. Because 64-bit counters can't practically overflow in observable perspective.